tests/test_query.py

```diff
@@ -0,0 +1,54 @@
+import pandas as pd
+import numpy.testing as npt
+import pytest
+
+from bluepysnap import BluepySnapError
+from bluepysnap.query import resolve_ids, _positional_mask, _circuit_mask
+
+
+def test_positional_mask():
+    data = pd.DataFrame(range(3))
+    npt.assert_array_equal(_positional_mask(data, [1, 2]), [False, True, True])
+    npt.assert_array_equal(_positional_mask(data, [0, 2]), [True, False, True])
+
+
+def test_population_mask():
+    data = pd.DataFrame(range(3))
+    queries, mask = _circuit_mask(data, "default", {"population": "default", "other": "val"})
+    assert queries == {"other": "val"}
+    npt.assert_array_equal(mask, [True, True, True])
+
+    queries, mask = _circuit_mask(data, "default", {"population": "unknown", "other": "val"})
+    assert queries == {"other": "val"}
+    npt.assert_array_equal(mask, [False, False, False])
+
+    queries, mask = _circuit_mask(data, "default",
+                                  {"population": "default", "node_id": [2], "other": "val"})
+    assert queries == {"other": "val"}
+    npt.assert_array_equal(mask, [False, False, True])
+
+    queries, mask = _circuit_mask(data, "default", {"other": "val"})
+    assert queries == {"other": "val"}
+    npt.assert_array_equal(mask, [True, True, True])
+
+
+def test_operator_mask():
+    data = pd.DataFrame([
```

Comment:
I would recommend adding nested "$and" / "$or" queries just to check if you do the operator queries as expected.

Reply:
I did it in the node_population.ids() I think but it would be nice to have that here too

Reply:
I had one on the line 47. Added another one below it. Please verify.

Reply:
oh sorry I missed it indeed my bad